src/piggui.rs

```diff
@@ -160,17 +162,28 @@ impl Application for Gpio {
                 self.chosen_layout = layout;
             }
             Message::ConfigLoaded((filename, config)) => {
-                self.config_filename = Some(filename);
+                self.config_filename = Some(filename.clone());
                 self.gpio_config = config.clone();
 
-                for (pin_number, pin_function) in config.configured_pins.iter() {
```

Comment:
If removing this is the bug fix, is very self-contained and better to move to a separate small PR?